src/features/common/Layout/QueryParamsContext.tsx

```diff
@@ -53,38 +55,24 @@ const QueryParamsProvider: FC = ({ children }) => {
   }, [query.callback]);
 
   useEffect(() => {
-    if (
-      query.locale &&
-      appSupportedLocale.some(
-        (locale) => locale.key[0] + locale.key[1] === query.locale
-      )
-    ) {
-      setLanguage(query.locale);
-    } else {
-      if (localStorage?.getItem('language') === null) {
-        const userPreferenceLanguage =
-          navigator.language ?? navigator.languages;
+    if (localStorage.getItem('language') === null) {
+      const userBrowserLanguage = navigator.language ?? navigator.languages;
+      // checks whether tenant supported locale matches the user browser preference locale
+      const languageMatched = tenantSupportedLocale.find((locale) => {
+        return locale === userBrowserLanguage;
```

Comment:
This does not work as navigator.language can return `de-DE` and navigator.languages an array of locales, compare  https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages.

Try something like 
```
const userBrowserLanguage = (navigator.language && navigator.language.substring(0,2)) ?? (navigator.languages && navigator.languages[0].substring(0,2));
```